x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java

```diff
@@ -631,6 +631,44 @@ public void expandGroupIds(List<String> groupIds,  ActionListener<Set<String>> l
                 , client::search);
     }
 
+    /**
+     * Check if a group exists, that is there exists a job that is a member of
+     * the group. If there are one or more jobs that define the group then
+     * the listener responds with true else an error.
+     *
+     * @param groupId The group Id
+     * @param listener Returns true or failure
+     */
+    public void groupExists(String groupId, ActionListener<Boolean> listener) {
```

Comment:
I think that for consistency with `jobExists` the listener should get `Boolean.FALSE` if the group does not exist and it is up to the caller to know if they want to error on `false` or not. Throwing an error here does not seem right as no error actually occurred.

Reply:
Yes that provides a better interface for users I pushed the change.